projects/rocprofiler-systems/source/lib/rocprof-sys/library/cpu_freq.cpp

```diff
@@ -195,41 +185,27 @@ metadata_initialize_cpu_freq_pmc(size_t dev_id)
           COMPONENT, TIME, rocprofsys::trace_cache::ABSOLUTE, BLOCK, EXPRESSION, 0, 0 });
 }
 
-void
-rocpd_process_cpu_usage_events(const uint32_t device_id, uint64_t timestamp,
-                               const component::cpu_freq& freq, double mem_page,
-                               double virt_mem_page, double peak_mem,
-                               double context_switch, double page_fault, double user_time,
-                               double kernel_time)
+std::vector<uint8_t>
+serialize_freqs(const component::cpu_freq& freq)
 {
-    auto&       data_processor   = get_data_processor();
-    const auto* _name            = trait::name<category::cpu_freq>::value;
-    auto        name_primary_key = data_processor.insert_string(_name);
-    auto        event_id         = data_processor.insert_event(name_primary_key, 0, 0, 0);
-
-    auto& agent_mngr = agent_manager::get_instance();
-    auto base_id = agent_mngr.get_agent_by_type_index(device_id, agent_type::CPU).base_id;
-
-    auto insert_event_and_sample = [&](const char* name, double value) {
-        data_processor.insert_pmc_event(event_id, base_id, name, value);
-        data_processor.insert_sample(name, timestamp, event_id);
-    };
-
-    do_for_enabled_cpus([&](size_t cpu_id) {
-        insert_event_and_sample(
-            get_cpu_freq_track_name<category::cpu_freq>(cpu_id).c_str(), freq.at(cpu_id));
+    constexpr size_t idx_elements   = sizeof(size_t) / sizeof(uint8_t);
+    constexpr size_t value_elements = sizeof(float) / sizeof(uint8_t);
+
+    std::vector<uint8_t> result;
+    const auto enabled_cpus_size = component::cpu_freq::get_enabled_cpus().size();
+    const auto result_size       = enabled_cpus_size * (idx_elements + value_elements);
+    result.resize(result_size);
+    result.assign(result_size, 0);
+
+    size_t offset = 0;
+    do_for_enabled_cpus([&](const auto& idx) {
+        auto value                                         = freq.at(idx);
+        *reinterpret_cast<size_t*>(result.data() + offset) = idx;
+        offset += sizeof(size_t);
+        *reinterpret_cast<float*>(result.data() + offset) = value;
+        offset += sizeof(float);
```

Comment:
```suggestion
        auto value = freq.at(idx);
        std::memcpy(result.data() + offset, &idx, sizeof(size_t));
        offset += sizeof(size_t);
        std::memcpy(result.data() + offset, &value, sizeof(float));
        offset += sizeof(float);
```
Consider using `memcpy` rather than `reinterpret_cast`.